taskscheduler/scheduler.go

```diff
@@ -0,0 +1,38 @@
+package taskscheduler
+
+import (
+	"errors"
+	"sync"
+)
+
+// Scheduler manages the scheduling, execution, and reporting of tasks with
+// controlled concurrency.
+type Scheduler struct {
+	taskChan              chan taskHandler
+	doneChan              chan struct{}
+	wg                    sync.WaitGroup
+	concurrentWorkerLimit int
+}
+
+// NewScheduler creates and returns a Scheduler with specified taskChannelBuffer
```

Comment:
WDYT about this rephrasing?

```suggestion
// NewScheduler creates and returns a Scheduler with specified configurations of the size of tasks channel
// and the concurrent worker limit. For invalid values an error will be returned.
```

Reply:
WDYT about 
```
// NewScheduler creates and returns a Scheduler with the specified task channel size...
``` 
?
I prefer to avoid 2x "of".
@arieltod